File: Modules/MovieDB/Movie/MovieForm.cs


namespace MovieTutorial.MovieDB.Forms;

[FormScript("MovieDB.Movie")]
[BasedOnRow(typeof(MovieRow))]
public class MovieForm
{
    [TextAreaEditor(Rows = 3)]
    public string Title { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    [DisplayName("Cast"),MovieCastEditor,IgnoreName]
    public List<MovieCastRow> CastList { get; set; }
    [TextAreaEditor(Rows = 8)]
    public string Storyline { get; set; }
    public int Year { get; set; }
    [TextAreaEditor(Rows = 8)]
    public DateTime ReleaseDate { get; set; }
    public int Runtime { get; set; }
    public MovieKind Kind { get; set; }
    public List<int> GenreList { get; set; }

    public string PrimaryImages { get; set; }
    public string GalleryImages { get; set; }
}

File: Modules/ServerTypes/MovieDB/MovieForm.ts
import { MovieCastEditor } from "@/MovieDB/MovieCast/MovieCastEditor";
import { TextAreaEditor, IntegerEditor, EnumEditor, LookupEditor, MultipleImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { MovieKind } from "./MovieKind";

export interface MovieForm {
    Title: TextAreaEditor;
    Description: TextAreaEditor;
    CastList: MovieCastEditor;
    Storyline: TextAreaEditor;
    Year: IntegerEditor;
    ReleaseDate: TextAreaEditor;
    Runtime: IntegerEditor;
    Kind: EnumEditor;
    GenreList: LookupEditor;
    PrimaryImages: MultipleImageUploadEditor;
    GalleryImages: MultipleImageUploadEditor;
}

export class MovieForm extends PrefixedContext {
    static readonly formKey = 'MovieDB.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = TextAreaEditor;
            var w1 = MovieCastEditor;
            var w2 = IntegerEditor;
            var w3 = EnumEditor;
            var w4 = LookupEditor;
            var w5 = MultipleImageUploadEditor;

            initFormType(MovieForm, [
                'Title', w0,
                'Description', w0,
                'CastList', w1,
                'Storyline', w0,
                'Year', w2,
                'ReleaseDate', w0,
                'Runtime', w2,
                'Kind', w3,
                'GenreList', w4,
                'PrimaryImages', w5,
                'GalleryImages', w5
            ]);
        }
    }
}

[MovieKind]; // referenced types